### PR TITLE
refactor(content,kumascript): convert to typescript

### PR DESCRIPTION
**Pull request by caugner**
## Summary

Converts `content/*` and `kumascript/*` to TypeScript.

~~⚠️ Blocked by:~~

- https://github.com/mdn/yari/pull/6993

### CODE REVIEW

**Comment by fiji-flo** [comment]
@caugner can you merge main and trigger a stage build from this branch?

**Comment by caugner** [comment]
> can you merge main and trigger a stage build from this branch?

Done, and running: https://github.com/mdn/yari/actions/runs/3046564665/

**Comment by caugner** [comment]
@fiji-flo The build was successful, but was overwritten by the scheduled one from last night.

**Review by fiji-flo — APPROVED** [review]
Looking good.

File: build/cli.ts [hunk 1]
```diff
@@ -100,11 +100,11 @@ export interface BuiltDocuments {
 }
 
 async function buildDocuments(
-  files = null,
+  files: string[] = null,
```
[the remark this answers is not in the document]
Comment (caugner) [reply build/cli.ts:103]: It just defines the parameter type, not the variable type. So you cannot pass `null`, but the variable can still be `null`.

**Pull request merged**